presto-orc/src/main/java/com/facebook/presto/orc/metadata/DwrfStripeCacheMode.java

```diff
@@ -19,14 +19,65 @@
 public enum DwrfStripeCacheMode
 {
     // no cache
-    NONE,
+    NONE(false, false),
 
     // cache contains stripe indexes
-    INDEX,
+    INDEX(true, false),
 
     // cache contains stripe footers
-    FOOTER,
+    FOOTER(false, true),
 
     // cache contains stripe indexes and footers
-    INDEX_AND_FOOTER;
+    INDEX_AND_FOOTER(true, true);
+
+    private final boolean hasIndex;
+    private final boolean hasFooter;
+
+    DwrfStripeCacheMode(boolean hasIndex, boolean hasFooter)
+    {
+        this.hasIndex = hasIndex;
+        this.hasFooter = hasFooter;
+    }
+
+    public boolean hasIndex()
+    {
+        return hasIndex;
+    }
+
+    public boolean hasFooter()
+    {
+        return hasFooter;
+    }
+
+    /**
+     * Get the position of an element in the cacheOffsets that can be used to
+     * get the offset for the index streams slice for a given stripe.
+     */
+    public int getIndexOffsetPosition(int stripeOrdinal)
+    {
+        switch (this) {
+            case INDEX:
+                return stripeOrdinal;
+            case INDEX_AND_FOOTER:
+                return stripeOrdinal * 2;
```

Comment:
Dumb question: why is it `stripeOrdinal * 2` here and `stripeOrdinal * 2 + 1` below for footer offset?

Reply:
If the mode is set to BOTH then offsets would be saved in pairs for each stripe: 
```
Stripe 0 index offset
Stripe 0 footer offset
...
Stripe 0 index offset   <-- stripe idx * 2 + 0
Stripe 0 footer offset  <-- stripe idx * 2 + 1
```